server/src/main/java/io/crate/analyze/expressions/ExpressionAnalyzer.java

```diff
@@ -506,75 +505,56 @@ protected Symbol visitFunctionCall(FunctionCall node, ExpressionAnalysisContext
 
         @Override
         protected Symbol visitSimpleCaseExpression(SimpleCaseExpression node, ExpressionAnalysisContext context) {
-            /* CASE caseOperand
-             *   WHEN whenOperand1 THEN result1
-             *   WHEN whenOperand2 THEN result2
-             *   ELSE default
-             * END
-             * ---
-             * operands: [eq(caseOperand, whenOperand1), eq(caseOperand, whenOperand2)]
-             * results:  [result1, result2]
-             */
-            List<WhenClause> whenClauses = node.getWhenClauses();
-            List<Symbol> operands = new ArrayList<>(whenClauses.size());
-            List<Symbol> results = new ArrayList<>(whenClauses.size());
-            Symbol caseOperand = convert(node.getOperand(), context);
+            var whenClauses = node.getWhenClauses();
+            var arguments = new ArrayList<Symbol>(whenClauses.size() * 2 + 2);
+
+            var defaultValue = node.getDefaultValue();
+            arguments.add(convert(BooleanLiteral.TRUE_LITERAL, context));
+
+            if (defaultValue == null) {
+                arguments.add(convert(NullLiteral.INSTANCE, context));
+            } else {
+                arguments.add(convert(defaultValue, context));
+            }
+
             for (WhenClause whenClause : whenClauses) {
-                Symbol whenOperand = convert(whenClause.getOperand(), context);
-                operands.add(allocateFunction(EqOperator.NAME, List.of(caseOperand, whenOperand), context));
-                results.add(convert(whenClause.getResult(), context));
+                arguments.add(buildCondition(node.getOperand(), whenClause.getOperand(), context));
```

Comment:
```suggestion
                Symbol condition = allocateFunction(
                    EqOperator.NAME,
                    List.of(convert(node.getOperand(), context), convert(whenClause.getOperand(), context)),
                    context
                );
                arguments.add(condition);
```

Just a suggestion and fine to keep as is.
The function is only used in one place and is more or less a single line. A local variable would convey the same semantics and reduce indirection. Answers the "what does buildCondition do?" while reading.